### PR TITLE
Release AWS 11.4.1

### PR DESCRIPTION
**Pull request by MarcelMue**
towards https://github.com/giantswarm/giantswarm/issues/11946

### CODE REVIEW

**Comment by whites11** [comment]
Please consider upgrading flatcar https://github.com/giantswarm/giantswarm/issues/11747

File: aws/v11.4.1/release.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    version: 1.16.0
+  - name: coredns
+    componentVersion: 1.6.5
+    version: 1.1.10
+  - name: external-dns
+    componentVersion: 0.7.2
+    version: 1.2.2
+  - name: kiam
+    componentVersion: 3.5.0
+    version: 1.2.2
```
Comment (njuettner) [inline aws/v11.4.1/release.yaml:29]: I updated the kiam-app to v1.3.0, see https://github.com/giantswarm/kiam-app/releases/tag/v1.3.0. I would be happy if we use in 11.4.1

```suggestion
    version: 1.3.0
```

Comment (MarcelMue) [reply aws/v11.4.1/release.yaml:29]: Fine if you can help with testing - this release has already gotten a bunch more changes. So if you could verify that it works, then fine by me.

Comment (njuettner) [reply aws/v11.4.1/release.yaml:29]: Sure, I'll help as much as I can 👍

**Comment by MarcelMue** [comment]
Used 
```
devctl create release --name 11.4.1 --provider aws --base 11.4.0 --app coredns@1.1.10@1.6.5 --app external-dns@1.2.2@0.7.2 --app kiam@1.3.0@3.5.0 --app kube-state-metrics@1.1.0@1.9.5 --app metrics-server@1.1.0@0.3.3 --app net-exporter@1.9.0 --component aws-operator@8.7.1-dev --component containerlinux@2512.2.1 --overwrite
```

for changelog now - still need to wait for aws-operator final testing & tagging

File: aws/v11.4.1/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# :zap: Giant Swarm Release v11.4.1 for AWS :zap:
+
+<< Add description here >>
```
Comment (tfussell) [inline aws/v11.4.1/README.md:3]: This release re-activates the recent AWS release of high-availability (HA) masters while fixing issues with certain OIDC configurations.

**Comment by sslavic** [comment]
@tfussell somehow recent changes removed change with deactivating 11.3.3 that I added. Do you know what happened?

**Comment by MarcelMue** [comment]
> @tfussell somehow recent changes removed change with deactivating 11.3.3 that I added. Do you know what happened?

That was me - I thought it was from an auto generation mistake I did.
Will add back.

**Comment by sslavic** [comment]
Tested on giraffe:
- 11.4.1 cluster creation
- upgrade from 11.3.3 to 11.4.1

App upgrades worked well.

Noticed single issue only - in 11.4.1 cluster creation, one master node came up without master role associated. Maybe it took longer due to china, or cluster being HA. k8s-label-node service retries/timeout should be increased. Anyway IMO this is a minor, not a blocker - it affects new clusters only, recovery is easy. Lets create a ticket not to forget to increase the timeout.

```
{⎈ |giantswarm-zvh84:default} ➜  ~ k get nodes
NAME                                        STATUS   ROLES    AGE     VERSION
ip-10-1-0-113.cn-north-1.compute.internal   Ready    <none>   9m6s    v1.16.9
ip-10-1-0-116.cn-north-1.compute.internal   Ready    master   9m6s    v1.16.9
ip-10-1-0-60.cn-north-1.compute.internal    Ready    master   9m6s    v1.16.9
ip-10-1-1-179.cn-north-1.compute.internal   Ready    worker   8m17s   v1.16.9
ip-10-1-1-70.cn-north-1.compute.internal    Ready    worker   8m17s   v1.16.9
ip-10-1-1-98.cn-north-1.compute.internal    Ready    worker   8m17s   v1.16.9
```

```
giantswarm@ip-10-1-0-113 ~ $ systemctl status k8s-label-node.service
● k8s-label-node.service - Adds labels to the node after kubelet startup
   Loaded: loaded (/etc/systemd/system/k8s-label-node.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2020-07-07 18:29:58 UTC; 19min ago
  Process: 3315 ExecStart=/bin/sh -c    while [ "$(kubectl get nodes $(hostname | tr [:upper:] [:lower:])| wc -l)" -lt "1" ]; do echo "Waiting for healthy k8s" && sleep 20s;done;    kubectl>
 Main PID: 3315 (code=exited, status=1/FAILURE)
```

```
giantswarm@ip-10-1-0-113 ~ $ sudo journalctl -u k8s-label-node.service -f
-- Logs begin at Tue 2020-07-07 18:10:21 UTC. --
Jul 07 18:27:07 ip-10-1-0-113.cn-north-1.compute.internal sh[3315]: Waiting for healthy k8s
Jul 07 18:28:17 ip-10-1-0-113.cn-north-1.compute.internal sh[3315]: Unable to connect to the server: net/http: TLS handshake timeout
Jul 07 18:28:17 ip-10-1-0-113.cn-north-1.compute.internal sh[3315]: Waiting for healthy k8s
Jul 07 18:29:18 ip-10-1-0-113.cn-north-1.compute.internal sh[3315]: error: the server doesn't have a resource type "nodes"
Jul 07 18:29:18 ip-10-1-0-113.cn-north-1.compute.internal sh[3315]: Waiting for healthy k8s
Jul 07 18:29:48 ip-10-1-0-113.cn-north-1.compute.internal sh[3315]: Unable to connect to the server: net/http: TLS handshake timeout
Jul 07 18:29:58 ip-10-1-0-113.cn-north-1.compute.internal sh[3315]: Unable to connect to the server: net/http: TLS handshake timeout
Jul 07 18:29:58 ip-10-1-0-113.cn-north-1.compute.internal systemd[1]: k8s-label-node.service: Main process exited, code=exited, status=1/FAILURE
Jul 07 18:29:58 ip-10-1-0-113.cn-north-1.compute.internal systemd[1]: k8s-label-node.service: Failed with result 'exit-code'.
Jul 07 18:29:58 ip-10-1-0-113.cn-north-1.compute.internal systemd[1]: Failed to start Adds labels to the node after kubelet startup.
```

Restarting the service did the trick to recover.

```
giantswarm@ip-10-1-0-113 ~ $ systemctl status k8s-label-node.service
● k8s-label-node.service - Adds labels to the node after kubelet startup
   Loaded: loaded (/etc/systemd/system/k8s-label-node.service; enabled; vendor preset: enabled)
   Active: active (exited) since Tue 2020-07-07 18:55:03 UTC; 4s ago
  Process: 26207 ExecStart=/bin/sh -c    while [ "$(kubectl get nodes $(hostname | tr [:upper:] [:lower:])| wc -l)" -lt "1" ]; do echo "Waiting for healthy k8s" && sleep 20s;done;    kubect>
 Main PID: 26207 (code=exited, status=0/SUCCESS)
```

File: aws/v11.4.1/README.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
+# :zap: Giant Swarm Release v11.4.1 for AWS :zap:
+
+This release re-activates the recent AWS release of high-availability (HA) masters while fixing issues with certain OIDC configurations.
```
Comment (MartaD) [inline aws/v11.4.1/README.md:3]: Can you add link to release notes with HA Masters details?

Comment (MarcelMue) [reply aws/v11.4.1/README.md]: Done.

File: aws/v11.4.1/release.diff [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  - componentVersion: 0.18.1                                         - componentVersion: 0.18.1
+    name: node-exporter                                                name: node-exporter
+    version: 1.2.0                                                     version: 1.2.0
+  components:                                                        components:
+  - name: app-operator                                               - name: app-operator
+    version: 1.0.0                                                     version: 1.0.0
+  - name: aws-cni                                                    - name: aws-cni
+    version: 1.6.0                                                     version: 1.6.0
+  - name: aws-operator                                               - name: aws-operator
+    version: 8.7.0                                              |      version: 8.7.1
```
Comment (xh3b4sd) [inline aws/v11.4.1/release.diff:46]: Me seeing the diff feature the first time in action. 

woah_rage_face

File: aws/v11.4.1/release.diff [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  - name: calico                                                     - name: calico
+    version: 3.10.4                                                    version: 3.10.4
+  - name: cert-operator                                              - name: cert-operator
+    version: 0.1.0                                                     version: 0.1.0
+  - name: cluster-operator                                           - name: cluster-operator
+    version: 2.3.0                                                     version: 2.3.0
+  - name: containerlinux                                             - name: containerlinux
+    version: 2345.3.1                                           |      version: 2512.2.1
```
Comment (stone-z) [inline aws/v11.4.1/release.diff:54]: <3

**Comment by MarcelMue** [comment]
For completeness I tested on ginger:
- cluster creation of `11.4.1`
- cluster upgrades of `11.4.0` -> `11.4.1`

Didn't see any issues.

File: aws/v11.4.1/README.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
+# :zap: Giant Swarm Release v11.4.1 for AWS :zap:
+
+This release re-activates the recent AWS [release of high-availability (HA) masters](https://github.com/giantswarm/releases/tree/master/aws/v11.4.0) while fixing issues with certain OIDC configurations.
```
Comment (cokiengchiara) [inline aws/v11.4.1/README.md:3]:
```suggestion
This release re-activates the recent AWS [release of high-availability (HA) masters](https://github.com/giantswarm/releases/tree/master/aws/v11.4.0), fixing OIDC configurations issues.
```

Very minor and subjective suggestion to make it clearer that we fixed OIDC thingies _in order to_ reactivate the release.

File: aws/v11.4.1/README.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
```
Comment (cokiengchiara) [inline aws/v11.4.1/README.md:4]:
```suggestion
**Note for Solution Engineers:** This release contains an external-dns fix introduced in
[11.3.2](https://github.com/giantswarm/releases/blob/master/aws/v11.3.2/release-notes.md).
It requires manual intervention for cluster upgrades in China to work. When upgrading a
cluster, existing ingress `A+TXT` record sets do not get replaced with `CNAME+TXT` record sets
even when external-dns is configured with CNAMEs as preferred. After upgrading, delete the
ingress `A+TXT` record sets. external-dns will then automatically create `CNAME+TXT` record
sets.
```

@sslavic , I know you hate it. But I think we still need this note to persist here? :P Please confirm.

Comment (MarcelMue) [reply aws/v11.4.1/README.md:4]: I just put it in and merge here - okay? 
I kind of want to be done with this :sweat_smile:

**Pull request merged**